lib/tasks/change_history.rake

```diff
@@ -0,0 +1,67 @@
+namespace :change_history do
+  desc "Show all change notes for a document"
+  task :show, %i[content_id locale] => :environment do |_, args|
+    Document
+      .find_by(content_id: args.content_id, locale: args.locale || "en")
+      .change_history
+      .sort_by { |note| note["public_timestamp"] }
+      .reverse
+      .each do |entry|
+      puts [entry["id"], entry["public_timestamp"], entry["note"]].join(" | ")
+    end
+  end
+
+  desc "Delete a single change note for a document"
+  task :delete, %i[content_id change_history_id locale] => :environment do |_, args|
+    args.with_defaults(locale: "en")
+    change_history = Document.find_by(content_id: args.content_id, locale: args.locale).change_history
+    change_history_entry = change_history.find { |e| e["id"] == args.change_history_id }
+
+    raise "Change history #{args.change_history_id} does not exist" unless change_history_entry
+
+    change_history.delete(change_history_entry)
+    ChangeHistory::UpdateInteractor.call(
+      document: [args.content_id, args.locale].join(":"),
+      change_history: change_history,
+      user: user,
+    )
+  end
+
+  desc "Edit a single change note for a document"
+  task :edit, %i[content_id change_history_id note locale] => :environment do |_, args|
+    args.with_defaults(locale: "en")
+    change_history = Document.find_by(content_id: args.content_id, locale: args.locale).change_history
+    change_history_entry = change_history.find { |e| e["id"] == args.change_history_id }
+
+    raise "Change history #{args.change_history_id} does not exist" unless change_history_entry
+
+    change_history_entry["note"] = args.note
+    ChangeHistory::UpdateInteractor.call(
+      document: [args.content_id, args.locale].join(":"),
+      change_history: change_history,
+      user: user,
+    )
+  end
+
+  desc "Add a new change note for a document"
+  task :add, %i[content_id timestamp note locale] => :environment do |_, args|
+    args.with_defaults(locale: "en")
+    change_history = Document.find_by(content_id: args.content_id, locale: args.locale).change_history
+
+    change_history.prepend(
```

Comment:
Totally my fault, but I'm rethinking that we'd want to prepend this and now actually think we should validate the order of change history (after being bitten a fair amount yesterday with differences) a consequence would be that we'd need to sort these again after inserting.

Reply:
Updated this to append the new entry, and changed the interactor (or whatever we replace it with) to sort the change notes before updating the revision.